core/saltfilter/filter.go

```diff
@@ -0,0 +1,67 @@
+// Package saltfilter used for reuse detection ONLY
+package saltfilter
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// Those suggest value are all set according to
+// https://github.com/shadowsocks/shadowsocks-org/issues/44#issuecomment-281021054
+const (
+	SuggestEntries            = int(1e6)
+	SuggestFalsePositiveRate  = 1e-6
+	defaultBloomRingSlotCount = 5
```

Comment:
And why is this not exported?

Reply:
due to not expose too many controll variables to users.But now change to implements as libev dose(PingPongBF) so this variable removed